fail2ban/tests/actiontestcase.py

```diff
@@ -389,6 +389,51 @@ def testExecuteActionStartEmpty(self):
 		self.assertLogged('Nothing to do')
 		self.pruneLog()
 
+	def testExecuteWithVars(self):
+		self.assertTrue(self.__action.executeCmd(
+			r'''printf %b "foreign input:\n'''
+			r''' -- $f2bV_A --\n'''
+			r''' -- $f2bV_B --\n'''
+			r''' -- $(echo -n $f2bV_C) --''' # echo just replaces \n to test it as single line
+			r'''"''', 
+			varsDict={
+			'f2bV_A': 'I\'m a hacker; && $(echo $f2bV_B)', 
```

Comment:
This was my first thought, but the goal is:
- to rewrite escape tomorrow with "compiled" version, so replace only once and supply always as parameters dictionary (in this case no matter what the "hacker" will provide in the user input).
- myself can better analyze it if not replaced with placeholder;
- in my new ban-time-incr version I've introduced a **factor** for banning (so e. g. we can analyze and increase the factor to ban faster and longer if such unescaped chars present); Of course it is possible with the counting of the placeholder, but it make this confuse-able, bad readable and more complicated by searching for possible complex injections, etc.

IMHO as option it is imaginable.